libenv/sysinfo.c

```diff
@@ -1129,7 +1129,14 @@ static void OSClasses(EvalContext *ctx)
     {
         Xen_Domain(ctx);
     }
-
+#ifdef XEN_CPUID_SUPPORT
+    else if (Xen_Hv_Check())
```

Comment:
Why `else` ? Do you want to skip this check in case `/proc/xen/capabilities` exist?

According to [this](https://www.centos.org/docs/5/html/5.2/Virtualization/sect-Virtualization-Tips_and_tricks-Identifying_guest_type_and_implementation.html) the file might exist in both DomU and Dom0.